### PR TITLE
change status order

### PR DESCRIPTION
**Pull request by ghost**
## Description
change status order
## Reference Requirements
- Task ID: TI-1

## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

```
mysql> EXPLAIN select count(*) as aggregate from `orders` \G;
*************************** 1. row ***************************
           id: 1
  select_type: SIMPLE
        table: orders
   partitions: NULL
         type: index
possible_keys: NULL
          key: orders_user_id_foreign
      key_len: 4
          ref: NULL
         rows: 35
     filtered: 100.00
        Extra: Using index
1 row in set, 1 warning (0.03 sec)

ERROR: 
No query specified
```
```
mysql> EXPLAIN select * from `orders` limit 10 offset 0 \G;
*************************** 1. row ***************************
           id: 1
  select_type: SIMPLE
        table: orders
   partitions: NULL
         type: ALL
possible_keys: NULL
          key: NULL
      key_len: NULL
          ref: NULL
         rows: 35
     filtered: 100.00
        Extra: NULL
1 row in set, 1 warning (0.01 sec)

ERROR: 
No query specified
```
```
mysql> EXPLAIN select * from `users` where `users`.`id` in ('1', '4', '5', '7', '9', '13', '17') and `users`.`deleted_at` is null \G;
*************************** 1. row ***************************
           id: 1
  select_type: SIMPLE
        table: users
   partitions: NULL
         type: range
possible_keys: PRIMARY
          key: PRIMARY
      key_len: 4
          ref: NULL
         rows: 7
     filtered: 10.00
        Extra: Using where
1 row in set, 1 warning (0.01 sec)

ERROR: 
No query specified
```

### CODE REVIEW

File: app/Http/Controllers/Admin/OrdersController.php [hunk 1]
```diff
@@ -17,4 +17,20 @@ public function index()
...
+     *
+     * @param \Illuminate\Http\Request $request request
+     * @param \App\Models\order        $order   order
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function updateStatus(Request $request, Order $order)
+    {
+        $order['status'] = $request->status;
+        // dd($order['status']);
```
Comment (at-dungle) [inline app/Http/Controllers/Admin/OrdersController.php]: Xóa code test

File: resources/lang/en/order.php [hunk 2]
```diff
@@ -13,6 +13,10 @@
                 'date' => 'Date',
                 'status' => 'Status',
                 'action' => 'Action',
+                'confirm_status' => 'Are your sure change status order?',
```
Comment (at-dungle) [inline resources/lang/en/order.php]: you or your?

File: resources/views/admin/order/index.blade.php [hunk 3]
```diff
@@ -4,10 +4,10 @@
   <div class="content-wrapper">
     <!-- Content Header (Page header) -->
     <section class="content-header">
-      <h1>{{__('order.admin.index.list_order')}}</h1>
+      <h1>{{ __('order.admin.index.list_order') }}</h1>
       <ol class="breadcrumb">
-        <li><a href="{{route('admin')}}"><i class="fa fa-dashboard"></i> {{__('admin.dashboard')}}</a></li>
-        <li class="active">{{__('admin.manage_order')}}</li>
+        <li><a href="{{route('admin')}}"><i class="fa fa-dashboard"></i> {{ __('admin.dashboard') }}</a></li>
```
Comment (at-dungle) [inline resources/views/admin/order/index.blade.php]: thêm space chổ route

File: app/Http/Controllers/Admin/OrdersController.php [hunk 4]
```diff
@@ -17,4 +17,19 @@ public function index()
...
+     * Update status of order
+     *
+     * @param \Illuminate\Http\Request $request request
+     * @param \App\Models\order        $order   order
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function updateStatus(Request $request, Order $order)
+    {
+        $order['status'] = $request->status;
```
Comment (at-thinhnguyen) [inline app/Http/Controllers/Admin/OrdersController.php]: validator status nữa chớ

File: app/Http/Controllers/Admin/OrdersController.php [hunk 5]
```diff
@@ -17,4 +17,19 @@ public function index()
...
+        $order->save();
```
Comment (at-thinhnguyen) [inline app/Http/Controllers/Admin/OrdersController.php]: check lỗi nữa nè

**Review by ghost — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Admin/OrdersController.php [hunk 6]
```diff
@@ -17,4 +18,23 @@ public function index()
...
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function updateStatus(ChangeStatusRequest $request, Order $order)
+    {
+        try {
+            $order['status'] = $request->status;
+            $order->save();
+        } catch (Exception $e) {
+            flash(trans('message.post.fail_delete'))->error();
```
Comment (at-thinhnguyen) [inline app/Http/Controllers/Admin/OrdersController.php]: cái này dùng ajjax gửi lên, trả về flash dưới đó xử lý sao?

**Comment by at-dungle** [comment]
LGTM

**Review by ghost — CHANGES_REQUESTED** [review]

File: resources/views/admin/order/index.blade.php [hunk 7]
```diff
@@ -73,3 +69,6 @@
     <!-- /.content -->
   </div>
 @endsection
+@section('status')
+  <script src="js/updateStatusBorrow.js"></script>
```
Comment (at-thinhnguyen) [inline resources/views/admin/order/index.blade.php]: dùng asset

File: app/Http/Requests/ChangeStatusRequest.php [hunk 8]
```diff
@@ -24,7 +26,10 @@ public function authorize()
     public function rules()
     {
         return [
-            'status' => 'required|integer|exists:orders,status',
+            'status' => [
+                'required',
+                Rule::in([Order::PENDING, Order::ACCEPTED, Order::REJECTED]),
```
Comment (at-thinhnguyen) [inline app/Http/Requests/ChangeStatusRequest.php]: j khổ vậy, `'required|integer|in:'.implode(',',[Order::PENDING, Order::ACCEPTED, Order::REJECTED])`

**Comment by at-dungle** [comment]
LGTM

**Review by ghost — APPROVED** [review]

**Pull request merged**